source/github-helpers/parse-route.ts

```diff
@@ -8,10 +8,17 @@ interface Pathname {
 	filePath: string;
 	toString: () => string;
 }
+
+const timeStampRegex = /.*@{(\d{4}([.\-/ ])\d{2}\2\d{2}T\d\d:\d\d:\d\dZ)}/;
+function isTimeStamp(part1: string): boolean {
+	return timeStampRegex.test(decodeURIComponent(part1));
+}
+
 export default function parseRoute(pathname: string): Pathname {
 	const [user, repository, route, ...next] = pathname.replace(/^\/|\/$/g, '').split('/');
-	const parts = next.join('/');
 	const branch = getCurrentBranch();
+	next[0] = isTimeStamp(next[0]) ? branch + '/' : next[0];
+	const parts = next.join('/');
 	if (parts !== branch && !parts.startsWith(branch + '/')) {
```

Comment:
Maybe 
```suggestion
	if (!isTimeStamp(next[0]) && parts !== branch && !parts.startsWith(branch + '/')) {
```

Reply:
That will only work if there is no file name.
https://github.com/sindresorhus/eslint-plugin-unicorn/tree/HEAD@%7B2020-05-24T16:33:08Z%7D

But the following will not
https://github.com/sindresorhus/refined-github/blob/master@%7B2019-04-20T03:27:17Z%7D/license
https://github.com/sindresorhus/refined-github/blob/master@%7B2019-12-11T10:12:11Z%7D/source/features/comments-time-machine-links.tsx

Reply:
Just this

```suggestion
	// Exclude extended revisions from being tested #3137 https://git-scm.com/docs/git-rev-parse#_specifying_revisions
	if (parts !== branch && !parts.startsWith(branch + '/') && !/\^|~|@{/.test(next[0])) {
```